### PR TITLE
Support aten::binary_cross_entropy_with_logits

### PR DESCRIPTION
**Pull request by Jyc323**
### Details:
 - updated op_table.cpp
 - implemented binary_cross_entropy_with_logits.cpp
 - implemented the unit test in test_binary_cross_entropy_with_logits.py

### Tickets:
 - *#29690*

### CODE REVIEW

**Review by Rishav0907 — COMMENTED** [review]
Directly computing the loss like this: - (target * log_sigmoid + (1 - target) * log(1 - sigmoid)) may lead to numerical instability

**Comment by Jyc323** [comment]
@Rishav0907, thanks for your suggestion! I have replaced the sigmoid calculation with using logits directly in a numerically stable formulation. Here is the equation I applied:

$$
\text{BCEWithLogits}(x, y) = \max(x, 0) - x \cdot y + \log\left(1 + \exp(-|x|)\right)
$$

Please review, and let me know if there is any issue. Thank you!

**Comment by mlukasze** [comment]
ping @mvafin || @mmikolajcz

**Comment by Jyc323** [comment]
@mlukasze @mvafin @mmikolajcz based on the failed checks, I modified cpp file to support 5 inputs, and fixed clang-format error. Please review and let me know if there is any issue or suggestion! Thank you!
